### PR TITLE
fix bcftools mpileup

### LINKED ISSUE #1104: Don't run samtools mpileup when only tools mpileup is selected
### Description of the bug

currently samtools mielup and bcftools mpileup are both run when `tools mpileup ` is selected, however only run bcftools then + fix the output docs if this is done

### Command used and terminal output

_No response_

### Relevant files

_No response_

### System information

_No response_

### PR DESCRIPTION
**Pull request by FriederikeHanssen**
Close #1104 #1103 #1099 #835 

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: subworkflows/local/bam_variant_calling_mpileup/main.nf [hunk 1]
```diff
@@ -1,7 +1,7 @@
-include { CAT_CAT as CAT_MPILEUP         } from '../../../modules/nf-core/cat/cat/main'
-include { BCFTOOLS_MPILEUP               } from '../../../modules/nf-core/bcftools/mpileup/main'
-include { SAMTOOLS_MPILEUP               } from '../../../modules/nf-core/samtools/mpileup/main'
-include { GATK4_MERGEVCFS                } from '../../../modules/nf-core/gatk4/mergevcfs/main'
+include { CAT_CAT as CAT_MPILEUP                    } from '../../../modules/nf-core/cat/cat/main'
+include { BCFTOOLS_MPILEUP                          } from '../../../modules/nf-core/bcftools/mpileup/main'
+include { SAMTOOLS_MPILEUP                          } from '../../../modules/nf-core/samtools/mpileup/main'
+include { GATK4_MERGEVCFS as MERGE_BCFTOOLS_MPILEUP } from '../../../modules/nf-core/gatk4/mergevcfs/main'
```
Comment (maxulysse) [inline subworkflows/local/bam_variant_calling_mpileup/main.nf:6]:
```suggestion
include { CAT_CAT         as CAT_MPILEUP           } from '../../../modules/nf-core/cat/cat/main'
include { BCFTOOLS_MPILEUP                          } from '../../../modules/nf-core/bcftools/mpileup/main'
include { SAMTOOLS_MPILEUP                          } from '../../../modules/nf-core/samtools/mpileup/main'
include { GATK4_MERGEVCFS as MERGE_BCFTOOLS_MPILEUP } from '../../../modules/nf-core/gatk4/mergevcfs/main'

```

**Review by maxulysse — APPROVED** [review]

File: subworkflows/local/bam_variant_calling_mpileup/main.nf (hunk 1 above)
Comment (maxulysse) [reply subworkflows/local/bam_variant_calling_mpileup/main.nf:6]: this is not needed, but I like having `as` aligned

Comment (FriederikeHanssen) [reply subworkflows/local/bam_variant_calling_mpileup/main.nf:6]: yes agree

File: tests/test_mpileup.yml [hunk 2]
```diff
@@ -52,7 +64,10 @@
       md5sum: f3dac01ea66b95fe477446fde2d31489
     - path: results/no_intervals.bed.gz.tbi
       md5sum: f3dac01ea66b95fe477446fde2d31489
-    - path: results/variant_calling/mpileup/sample1/sample1.normal.mpileup.gz
-    # binary changes md5sums on reruns
+    - path: results/variant_calling/bcftools/sample1/sample1.bcftools.vcf.gz
+      md5sum: 152f5925eef9a05f96a66b805ff7f931
+    - path: results/variant_calling/bcftools/sample1/sample1.bcftools.vcf.gz.tbi
+      md5sum: 277b1f3c0f32e80ed736d0d927c94a62
+    - path: results/variant_calling/mpileup/
```
Comment (FriederikeHanssen) [inline tests/test_mpileup.yml:71]:
```suggestion
    - path: results/variant_calling/mpileup/
      should_exist: false
```

**Comment by maxulysse** [comment]
> Close #1104 #1103 #1099 #835

How many issues are you closing with this one PR?

**Comment by FriederikeHanssen** [comment]
all of them, tbf 3 of  them are the same thing

File: tests/test_mpileup.yml [hunk 3]
```diff
@@ -5,8 +5,12 @@
     - mpileup
   files:
     - path: results/multiqc
+    - path: results/variant_calling/bcftools/sample2/sample2.bcftools.vcf.gz
+      md5sum: 07c3eb437131bfa5ec036e7da039ad2f
+    - path: results/variant_calling/bcftools/sample2/sample2.bcftools.vcf.gz.tbi
+      md5sum: d828e532551df740c98e9b1c0582cf31
```
Comment (maxulysse) [inline tests/test_mpileup.yml:11]:
```suggestion
    # binary changes md5sums on reruns
    - path: results/variant_calling/bcftools/sample2/sample2.bcftools.vcf.gz.tbi
    # binary changes md5sums on reruns
```

Comment (maxulysse) [reply tests/test_mpileup.yml:11]: missing these two

**Pull request merged**